### ISSUE TITLE
The `site` variable isn't present within Ruby front matter

### ISSUE THREAD

**Issue by jaredcwhite** [bridgetownrb/bridgetown#372]
Given Ruby front matter of this:

```ruby
front_matter do
  layout :default
  title site.metadata.podcast_title
end
```

It produces an error because the front matter is evaluated within an instance of `RepoOrigin` which doesn't provide a `site` accessor.

**Comment by jaredcwhite** [comment]
Perhaps one possible answer is to use the lambda functionality for later evaluation (as that is evaluated within an instance of the resource itself):

```
title ->{ site.metadata.podcast_title }
```

However, I'm getting an empty `{}` metadata in that case, so there's still a bug somewhere. :(

**Comment by jaredcwhite** [comment]
Ah, it's because the time the evaluation happens is still within the loop of reading collections and the data collection apparently hasn't been read and merged into the site data tree yet. We need to frontload the data collection getting read in before any other collection!

**Comment by jaredcwhite** [comment]
I think the lambda function along with the fix in #373 is sufficient to resolve this matter. Closing.
